src/battle_setup.c

```diff
@@ -1660,6 +1662,9 @@ static bool32 UpdateRandomTrainerRematches(const struct RematchTrainer *table, u
     s32 i;
     bool32 ret = FALSE;
 
+    if (CheckBagHasItem(ITEM_VS_SEEKER, 1) == TRUE)
+        return ret;
+
```

Comment:
Since we are touching that function I would personally like to see less indentations and we probably don't need to store the return variable but can use early returns